unic/ucd/block/src/block.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2012-2015 The Rust Project Developers.
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use unic_char_property::tables::CharDataTableIter;
+use unic_char_range::CharRange;
+
+#[derive(Debug)]
+pub struct Block {
+    pub range: CharRange,
+    pub name: &'static str,
+}
+
+impl Block {
+    pub fn of(chr: char) -> Option<Block> {
+        Block::new(data::BLOCKS.find_with_range(chr))
+    }
+
+    fn new(raw_block: Option<(CharRange, &'static str)>) -> Option<Block> {
+        match raw_block {
+            None => None,
```

Comment:
Hum... why do we need to be able to call `new()` with `None` and receive `None`. It's kind of counter-intuitive to Rust's `new()` pattern.

Reply:
Good point, I think I was trying optimize for the two use cases in this file which both provide an `Option<(CharRange, &'static str)>` and expect an `Option<Block>`.
Should be fixed now.
I have also added some comments about the missing surrogate code point blocks.